services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/TransactDeleteItemEnhancedRequest.java

```diff
@@ -37,14 +37,16 @@
  */
 @SdkPublicApi
 @ThreadSafe
-public final class TransactDeleteItemEnhancedRequest {
+public final class TransactDeleteItemEnhancedRequest<T> {
```

Comment:
This is backward incompatible change , just modified this for POC.
May be we can add a new class TransactDeleteCompleteItemEnhancedRequest or some good name and use this for specific case of optimistic delete case where we pass complete Items  instead of just key.